components/calendar/calendar.js

```diff
@@ -636,22 +641,22 @@ class Calendar extends LocalizeStaticMixin(LitElement) {
 	_updateShownMonthDecrease() {
 		if (this._shownMonth === 0) this._shownYear--;
 		this._shownMonth = getPrevMonth(this._shownMonth);
-		this.classList.remove('d2l-calendar-initial-month');
-		this.classList.remove('d2l-calendar-prev');
-		this.classList.remove('d2l-calendar-next');
+		this._initialMonth = false;
+		this._nextMonthNav = false;
+		this._prevMonthNav = false;
 		setTimeout(() => {
-			this.classList.add('d2l-calendar-prev');
+			this._prevMonthNav = true;
 		}, 100); // timeout for firefox
 	}
 
 	_updateShownMonthIncrease() {
 		if (this._shownMonth === 11) this._shownYear++;
 		this._shownMonth = getNextMonth(this._shownMonth);
-		this.classList.remove('d2l-calendar-initial-month');
-		this.classList.remove('d2l-calendar-prev');
-		this.classList.remove('d2l-calendar-next');
+		this._initialMonth = false;
+		this._nextMonthNav = false;
+		this._prevMonthNav = false;
 		setTimeout(() => {
```

Comment:
What happens in FF without the timeout?  Is the timeout still needed now that the DOM mutation has been moved to `render`?

Reply:
So with this change to use the classMap etc., the `setTimeout` seems to be needed for all browsers, and then firefox seems to need it to actually have a value or else basically every other month change transition just doesn't happen.

I have tried removing the setTimeout and using other things like `requestUpdate` + `await this.updateComplete` before modifying the class, moving code around and so far I haven't had luck avoiding the `setTimeout` so definitely open to ideas to avoid this.